src/app/route/ProductPage/ProductPage.container.js

```diff
@@ -79,10 +82,25 @@ export class ProductPageContainer extends PureComponent {
         this._onProductUpdate();
     }
 
-    componentDidUpdate({ location: { pathname: prevPathname } }) {
-        const { location: { pathname } } = this.props;
+    componentDidUpdate(prevProps) {
+        const {
+            location: { pathname },
+            product: { id }
+        } = this.props;
+        const {
+            location: { pathname: prevPathname },
+            product: { id: prevId }
+        } = prevProps;
 
         if (pathname !== prevPathname) this._requestProduct();
+
+        if (id !== prevId) {
+            const dataSource = this._getDataSource();
+            const { updateMetaFromProduct } = this.props;
+
+            updateMetaFromProduct(dataSource);
+        }
+
         this._onProductUpdate();
```

Comment:
Why not to use on product update? Is it not reliable?

Reply:
I went to magento admin to see if it is possible to set defferent meta for each configuration of configurable product and it turned out that it is not. 
Also went to SEO specialist for the same purpose. Was told to get single meta for configurable product.
Therefore it is necessary to update meta only on product change, what I did.